sql/core/src/test/scala/org/apache/spark/sql/execution/benchmark/UnsafeArrayDataBenchmark.scala

```diff
@@ -0,0 +1,256 @@
+    }
+
+    val doubleUnsafeArray = new UnsafeArrayData
+    val doubleSize = calculateHeaderPortionInBytes(count) + 8 * count
+    val doubleBuffer = new Array[Byte](doubleSize)
+    Platform.putInt(doubleBuffer, Platform.BYTE_ARRAY_OFFSET, count)
+    doubleUnsafeArray.pointTo(doubleBuffer, Platform.BYTE_ARRAY_OFFSET, doubleSize)
+    var doublePrimitiveArray: Array[Double] = null
+    val readDoubleArray = { i: Int =>
+      var n = 0
+      while (n < iters) {
+        doublePrimitiveArray = doubleUnsafeArray.toDoubleArray
+        n += 1
+      }
+    }
+
+    val benchmark = new Benchmark("Get primitive array from UnsafeArrayData", count * iters)
+    benchmark.addCase("Int")(readIntArray)
+    benchmark.addCase("Double")(readDoubleArray)
+    benchmark.run
+    /*
+    Java HotSpot(TM) 64-Bit Server VM 1.8.0_92-b14 on Mac OS X 10.10.4
+    Intel(R) Core(TM) i5-5257U CPU @ 2.70GHz
+
+    Get primitive array from UnsafeArrayData: Best/Avg Time(ms)    Rate(M/s)   Per Row(ns)  Relative
+    ------------------------------------------------------------------------------------------------
+    Int                                            100 /  176        632.1           1.6       1.0X
+    Double                                         267 /  334        236.0           4.2       0.4X
+    */
+  }
+
+  def putPrimitiveArray(iters: Int): Unit = {
+    val count = 1024 * 1024 * 12
+
+    val intPrimitiveArray: Array[Int] = new Array[Int](count)
+    var intUnsafeArray: UnsafeArrayData = null
+    val createIntArray = { i: Int =>
+      var n = 0
+      while (n < iters) {
+        intUnsafeArray = UnsafeArrayData.fromPrimitiveArray(intPrimitiveArray)
+        n += 1
+      }
+    }
+
+    val doublePrimitiveArray: Array[Double] = new Array[Double](count)
+    var doubleUnsafeArray: UnsafeArrayData = null
+    val createDoubleArray = { i: Int =>
+      var n = 0
+      while (n < iters) {
+        doubleUnsafeArray = UnsafeArrayData.fromPrimitiveArray(doublePrimitiveArray)
+        n += 1
+      }
+    }
+
+    val benchmark = new Benchmark("Create UnsafeArrayData from primitive array", count * iters)
+    benchmark.addCase("Int")(createIntArray)
+    benchmark.addCase("Double")(createDoubleArray)
+    benchmark.run
+    /*
+    Java HotSpot(TM) 64-Bit Server VM 1.8.0_92-b14 on Mac OS X 10.10.4
+    Intel(R) Core(TM) i5-5257U CPU @ 2.70GHz
+
+    Create UnsafeArrayData from primitive array: Best/Avg Time(ms)  Rate(M/s)  Per Row(ns)  Relative
+    ------------------------------------------------------------------------------------------------
+    Int                                             69 /  168        911.5           1.1       1.0X
+    Double                                         298 /  328        210.8           4.7       0.2X
+    */
+  }
+
+  ignore("Benchmark UnsafeArrayData") {
+    readUnsafeArray(10)
+    writeUnsafeArray(10)
+    getPrimitiveArray(5)
+    putPrimitiveArray(5)
+  }
+
+  def main(args: Array[String]): Unit = {
```

Comment:
```main()``` is not used actually. I will remove it. The steps are as follows:
1. Replace ```ignore(``` with ```test(```
2. Run this with ```build/sbt "sql/test-only *benchmark.UnsafeArrayDataBenchmark"```

I will add step 1 in the comment